dev-utils/src/commands/init_command.rs

```diff
@@ -0,0 +1,242 @@
+
+        {
+            let obj = self.genesis_dev.as_object_mut().unwrap();
+
+            // change
+            let old_genesis_time = obj["genesis_time"].as_str().unwrap().to_string();
+            let new_genesis_time: String = input()
+                .msg(format!("genesis_time( {} )=", old_genesis_time))
+                .default(old_genesis_time)
+                .get();
+            obj["genesis_time"] = json!(new_genesis_time);
+            // save
+            self.genesis_time = obj["genesis_time"].clone();
+            let councils = obj["council_nodes"].as_array_mut().unwrap();
+            councils.push(json!({"consensus_pubkey_type": "Ed25519",}));
+
+            councils[0]["staking_account_address"] = json!(self.staking_account_address);
+            councils[0]["consensus_pubkey_b64"] = json!(self.tendermint_pubkey.as_str());
+
+            obj["launch_incentive_from"] = json!(input()
+                .msg(format!(
+                    "launch_incentive_from({})=",
+                    self.distribution_addresses[0]
+                ))
+                .default(self.distribution_addresses[0].clone())
+                .get());
+            obj["launch_incentive_to"] = json!(input()
+                .msg(format!(
+                    "launch_incentive_to({})=",
+                    self.distribution_addresses[1]
+                ))
+                .default(self.distribution_addresses[1].clone())
+                .get());
+            obj["long_term_incentive"] = json!(input()
+                .msg(format!(
+                    "long_term_incentive({})=",
+                    self.distribution_addresses[2]
+                ))
+                .default(self.distribution_addresses[2].clone())
+                .get());
+        }
+        println!(
+            "read_information={}",
+            serde_json::to_string_pretty(&self.genesis_dev).unwrap()
+        );
+    }
+    pub fn generate_app_info(&mut self) {
+        let path = Path::new("./coin.json");
+
+        println!(
+            "{}",
+            serde_json::to_string_pretty(&self.genesis_dev).unwrap()
+        );
+        File::create(&path)
+            .map(|mut file| {
+                let note = serde_json::to_string(&self.genesis_dev).unwrap();
+                file.write_all(note.as_bytes()).unwrap();
+                ()
+            })
+            .unwrap();
+
+        println!("-----------------------------------------------");
+        // app_hash,  app_state
+        let result = GenesisCommand::generate(&path.to_path_buf()).unwrap();
+        println!("genesis_time( {} )=", self.genesis_time);
+        println!("-----------------------------------------------");
+        self.app_hash = json!(result.0);
+        self.app_state = serde_json::from_str(&result.1).unwrap();
+    }
+    pub fn get_tendermint_filename(&self) -> String {
+        format!(
+            "{}/.tendermint/config/genesis.json",
+            dirs::home_dir().unwrap().to_str().unwrap()
```

Comment:
is it always in home dir?

Reply:
yes, env::home_dir is deprecated, so I chose this one.